src/sinks/aws_s3/config.rs

```diff
@@ -32,7 +34,6 @@ const DEFAULT_FILENAME_TIME_FORMAT: &str = "%s";
 const DEFAULT_FILENAME_APPEND_UUID: bool = true;
 
 #[derive(Deserialize, Serialize, Debug, Clone)]
-#[serde(deny_unknown_fields)]
```

Comment:
Why did we drop this?

Reply:
I think I asked this on an earlier PR and the answer was that it no longer works with two `flatten` annotations on the same struct.

Reply:
Might be a good one to add to @tobz's list for the new config system.

Reply:
The underlying issue is this one: https://github.com/serde-rs/serde/issues/1600.

Deserialization fails when combining `deny_unknown_fields` and untagged enums, the `generate::tests::generate_all` and related tests fail with

```
vector generate::tests::generate_all ---

running 1 test
"//aws_s3" : unknown field `encoding` for key `sinks.sink0` at line 36 column 1
test generate::tests::generate_all ... FAILED
```

which means that no sink config that uses `EncodingConfigAdapter` can have `deny_unknown_fields` 😐

Some more context in https://github.com/vectordotdev/vector/issues/12162

Reply:
Aha, gotcha, we can plan to replace it when we remove the shim.